### PR TITLE
ref(ddm): Split utils

### PR DESCRIPTION
**Pull request by ArthurKnaus**
Split metric utils to reduce the dependencies that are being imported into different parts of the code base.
Added new files for `formatters`, `types` and `constants`.

### CODE REVIEW

File: static/app/views/dashboards/utils.tsx [hunk 1]
```diff
@@ -41,12 +41,9 @@ import {
 } from 'sentry/utils/discover/fields';
 import {DiscoverDatasets, DisplayModes} from 'sentry/utils/discover/types';
 import {getMeasurements} from 'sentry/utils/measurements/measurements';
-import {
-  getDdmUrl,
-  getMetricDisplayType,
-  MetricWidgetQueryParams,
-} from 'sentry/utils/metrics';
+import {getDdmUrl, getMetricDisplayType} from 'sentry/utils/metrics';
```
Comment (JonasBa) [inline static/app/views/dashboards/utils.tsx:44]: Should this fn be getMetricsUrl?

**Review by JonasBa — APPROVED** [review]
few non blocking comments, but I see that you prefix a lot of the naming with `Metrics` and a few are left either referencing `ddm` or are not metrics specific.

On another note, it seems like we are not inbetween ddm and metrics naming conventions, if you have some time I would love it if we can just rename everything into metrics so that we don't end up with dead concepts in the codebase. It would be a bit more friendlier to contributors and new folks joining us who wont know what ddm is :)

**Review by obostjancic — APPROVED** [review]

**Comment by ArthurKnaus** [comment]
@JonasBa yes, that is the plan. Doing it step by step. During prototyping and development things got mixed up a bit between ddm (the view) and metrics as a dataset. 
As this PR only moves code to different files, I will leave it now as it is. Will create a follow-up that prefixes the too generic exports.

**Comment by JonasBa** [comment]
> @JonasBa yes, that is the plan. Doing it step by step. During prototyping and development things got mixed up a bit between ddm (the view) and metrics as a dataset. As this PR only moves code to different files, I will leave it now as it is. Will create a follow-up that prefixes the too generic exports.

Yeah, makes sense. I think it's ok to have it in a transient state, but would be nice to just rename every ddm instance to metrics inside the codebase. If we don't, it'll just live forever in the codebase and since it's still early stages of the project, best do it now before it grows further

**Pull request merged**